architecture/src/commonMain/kotlin/observable/Observables.kt

```diff
@@ -0,0 +1,466 @@
+    val coroutineScope: CoroutineScope,
+    val context:CoroutineContext,
+    val flow:() -> Flow<T>,
+    val setter: suspend (T) -> Unit,
+    val poster: (T) -> Unit = { coroutineScope.launch(context) { setter(it) }},
+    observation: Observation<R, T, OO>
+    ):SuspendableSetter<T> {
+
+    init {
+        observeFlow(observation, coroutineScope, context, flow())
+    }
+
+    override suspend fun set(newValue: T) = setter(newValue)
+    override fun post(newValue: T) = poster(newValue)
+}
+
+class HotFlowableDefaultSubject<R:T?, T>(
+    defaultValue:R,
+    val coroutineScope: CoroutineScope,
+    context: CoroutineContext = Dispatchers.Main.immediate,
+    hotFlowable: HotFlowable<T?>,
+    observation: ObservationDefault<R, T?> = ObservationDefault(defaultValue, Value(hotFlowable.initialValue))
+) : BaseDefaultSubject<R, T?>(observation),
+    SuspendableSetter<T?> by MutableFlowSubjectHelper(
+        coroutineScope,
+        context,
+        flow = { hotFlowable.flow()},
+        setter = { hotFlowable.set( it )},
+        observation = observation
+    ) {
+    init {
+        bind(coroutineScope, context)
+    }
+}
+
+class HotFlowableInitializedSubject<T>(
+    val coroutineScope: CoroutineScope,
+    context: CoroutineContext = Dispatchers.Main.immediate,
+    hotFlowable: HotFlowable<T>,
+    observation: ObservationInitialized<T> = ObservationInitialized(Value(hotFlowable.initialValue))
+) : BaseInitializedSubject<T>(observation),
+    SuspendableSetter<T> by MutableFlowSubjectHelper(
+        coroutineScope,
+        context,
+        flow = { hotFlowable.flow()},
+        setter = { hotFlowable.set( it )},
+        observation = observation
+    ) {
+    init {
+        bind(coroutineScope, context)
+    }
+}
+
+
+/**
+ * [BaseSubject] that synchronizes its value to a [MutableStateFlow]
+ * @param observedStateFlow The [MutableStateFlow] to synchronize to
+ * @param coroutineScope The [CoroutineScope] on which to observe changes to the [StateFlow]
+ */
+open class StateFlowSubject<T>(
+    val coroutineScope: CoroutineScope,
+    val context:CoroutineContext = Dispatchers.Main.immediate,
+    private val observedStateFlow: MutableStateFlow<T>,
+    observation: ObservationInitialized<T> = ObservationInitialized(Value(observedStateFlow.value)),
+) :  BaseInitializedSubject<T>(observation),
+    SuspendableSetter<T> by MutableFlowSubjectHelper(
+        coroutineScope,
+        context,
+        { observedStateFlow },
+        { observedStateFlow.value = it },
+        { observedStateFlow.value = it},
+        observation) {
+    init {
+        bind(coroutineScope, context)
+    }
+}
+
+/**
+ * [BaseSubject] that synchronizes its value to a [MutableStateFlow]
+ * @param observedStateFlow The [MutableStateFlow] to synchronize to
+ * @param coroutineScope The [CoroutineScope] on which to observe changes to the [StateFlow]
+ */
+open class StateFlowDefaultSubject<R:T?, T>(
+    defaultValue: R,
```

Comment:
I know that is clear from the name, but these 2 classes have the same documentation. In `StateFlowDefaultSubject` you can add a `@param defaultValue` just to distinguish them

Reply:
This documentation is not even correct so I removed it. I want to go back and add a whole bunch more after it's been proven to work.